dataikuapi/dss/ml.py

```diff
@@ -627,46 +805,38 @@ class AlgorithmSettings(dict):
     def __init__(self, raw_settings, hyperparameter_search_params):
         super(AlgorithmSettings, self).__init__(raw_settings)
         self._hyperparameter_search_params = hyperparameter_search_params
+        self._hyperparameters_registry = dict()
+
+    def register_numerical_hyperparameter(self, name):
+        self._hyperparameters_registry[name] = NumericalHyperparameterSettings(name, self)
+        return self._hyperparameters_registry[name]
+
+    def register_categorical_hyperparameter(self, name):
+        self._hyperparameters_registry[name] = CategoricalHyperparameterSettings(name, self)
+        return self._hyperparameters_registry[name]
+
+    def register_single_valued_categorical_hyperparameter(self, name, accepted_values=None):
+        self._hyperparameters_registry[name] = SingleValuedCategoricalHyperparameterSettings(name, self, accepted_values=accepted_values)
+        return self._hyperparameters_registry[name]
+
+    def register_single_valued_hyperparameter(self, name, accepted_types=None):
+        self._hyperparameters_registry[name] = SingleValuedHyperparameterSettings(name, self, accepted_types=accepted_types)
+        return self._hyperparameters_registry[name]
 
     def _repr_html_(self):
-        active_settings = dict()
-        for key in self.keys():
-            if isinstance(self[key], dict):
-                # Searchable hyperparameter
-                if "range" in self[key]:
-                    # Numerical hyperparameter
-                    active_hyperparam = self._get_active_numerical_hyperparam_dict(key)
-                else:
-                    # Categorical hyperparameter
-                    active_hyperparam = self[key]
-                active_settings[key] = active_hyperparam
-            else:
-                active_settings[key] = self[key]
-        return "<pre>" + self.__class__.__name__ + "(active_values={})</pre>".format(json.dumps(active_settings, indent=4)) + \
-               "\n<details><pre>{}</pre></details>".format(self.__repr__())
+        res = "<pre>" + self.__class__.__name__ + "\n"
+        for name, hyperparam_settings in self._hyperparameters_registry.items():
+            res += hyperparam_settings._pretty_repr() + "\n"
+        res += "</pre>"
+        return res + "<details><pre>{}</pre></details>".format(self.__repr__())
 
     def __repr__(self):
         return self.__class__.__name__ + "(values={})".format(json.dumps(super(AlgorithmSettings, self).copy(), indent=4))
```

Comment:
Shouldn't we have a single line `__repr__` now that we have the `_repr_html_` ?

Reply:
Indeed, thanks !